reconcile/utils/jira_client.py

```diff
@@ -74,14 +80,19 @@ def create_issue(
         body: str,
         labels: Optional[Iterable[str]] = None,
         links: Iterable[str] = (),
+        issueType: Optional[Mapping[str, str]] = None,
+        assignee: Optional[str] = None,
     ) -> Issue:
         """Create an issue in our project with the given labels."""
+        if not issueType:
+            issueType = {"name": "Task"}
         issue = self.jira.create_issue(
             project=self.project,
             summary=summary,
             description=body,
             labels=labels,
-            issuetype={"name": "Task"},
+            issuetype=issueType,
+            assignee=assignee,
```

Comment:
I'm not sure if this line will work as expected.

Reply:
https://atlassian-python-api.readthedocs.io/jira.html#manage-issues

```
# Assign issue to user
jira.assign_issue(issue_key, account_id)
```

Reply:
Based on what I've read, project,summary,description,and labels are all required attributes for create_issue. All other fields are supported, including specifying an assignee.

https://community.atlassian.com/t5/Jira-Software-questions/Python-API-Create-issue-Assignee-field-is-not-populated/qaq-p/2014457

Reply:
I'm having trouble finding documentation for the `create_issue` function to understand if `assignee` is a valid parameter.

Assuming it is, my assumption from how I see "accounts" referenced in other function in the docs, you'll need to use a user's account_id, rather than a user's username.

Is that going to be okay with how this function is intended to be consumed?

Reply:
I would call this a "thin" client, as it doesn't concern itself too much with the arguments into the methods itself, it just passes them in to the API calls assuming you know what is supported and what is not. 

https://jira.readthedocs.io/examples.html#issues

I'm looking at the dict input examples as reference. 

And this note:

```
Project, summary, description and issue type are always required when creating issues. Your Jira may require additional fields for creating issues; see the jira.createmeta method for getting access to that information.
```

Off topic, but is that what a thin client really is? Or is there a definition?

Reply:
These are just FYI's to make sure you're aware of, in case this alters the way you want this particular MR to look.

Otherwise, there's nothing inherently wrong about this MR, and LGTM.

Reply:
Assigning based on username is supported from what I can tell. This support thread from Atlassian is not the best, but I used it as a data point giving me confidence that we can assign in "create_issue".

Reply:
This is pretty confused.

I'm reading directions like this that supposedly how to check your "account ID", but I don't actually see an account ID.

https://community.atlassian.com/t5/Jira-questions/how-to-find-accountid/qaq-p/1111436

---

Then, according to answers like this, Jira-on-prem doesn't use an account-id?

https://community.atlassian.com/t5/Jira-questions/How-to-find-account-id-on-jira-on-premise/qaq-p/1168652

So I guess that means we're using Jira-on-prem and we don't have account-ids?

In which case, I have no idea what you're supposed to actually use with this library for assignee. Hopefully using a "username" works after all because I guess that's all we can use lol.